pkg/disk/filesystem.go

```diff
@@ -9,47 +9,106 @@ import (
 	"github.com/google/uuid"
 )
 
-type MkfsOption int
+type MkfsOption interface {
+	json.Marshaler
+	isMkfsOption()
+	Clone() MkfsOption
+}
 
-const ( // MkfsOption type enum
-	MkfsVerity MkfsOption = iota // Enable fs-verity option if needed (typically for EXT4)
-)
+type MkfsOptionVerity struct{}
 
-func getMkfsOptionMapping() []string {
-	return []string{"verity"}
+func (MkfsOptionVerity) isMkfsOption()     {}
+func (MkfsOptionVerity) Clone() MkfsOption { return &MkfsOptionVerity{} }
+func (o MkfsOptionVerity) MarshalJSON() ([]byte, error) {
+	return json.Marshal("verity")
 }
 
-// String converts MkfsOption into a human readable string
-func (option MkfsOption) String() string {
-	return getMkfsOptionMapping()[int(option)]
+type MkfsOptionGeometry struct {
+	Heads           int `json:"heads" yaml:"heads"`
+	SectorsPerTrack int `json:"sectors-per-track" yaml:"sectors-per-track"`
```

Comment:
Hmm, this should maybe be sectors_per_track. The stage uses sectors-per-track, but we should probably match the other payload fields which seem to use underscore.